src/test/java/com/puppycrawl/tools/checkstyle/AbstractPathTestSupport.java

```diff
@@ -51,6 +54,19 @@ protected final String getPath(String filename) throws IOException {
                 .getCanonicalPath();
     }
 
+    protected static String getLineSeparatorForFile(String filepath, Charset charset)
```

Comment:
The older I become the more I dislike `protected`.
 
move it to Utils - https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/CheckUtil.java to make such context-less method/function to be accessible to others if required.